processor/src/main/resources/org/mapstruct/ap/internal/model/LifecycleCallbackMethodReference.ftl

```diff
@@ -18,4 +18,7 @@
      limitations under the License.
 
 -->
-<#if declaringType??>${instanceVariableName}.</#if>${name}(<#list parameterAssignments as param> <#if param.targetType><@includeModel object=ext.targetType raw=true/>.class<#elseif param.mappingTarget>${ext.targetBeanName}<#else>${param.name}</#if><#if param_has_next>,<#else> </#if></#list>);
+<#if hasReturnType()>${mappingTargetType} ${targetVariableName} = </#if><#if declaringType??>${instanceVariableName}.</#if>${name}(<#list parameterAssignments as param> <#if param.targetType><@includeModel object=ext.targetType raw=true/>.class<#elseif param.mappingTarget>${ext.targetBeanName}<#else>${param.name}</#if><#if param_has_next>,<#else> </#if></#list>);
```

Comment:
Use `<@includeModel object=mappingTargetType/>` instead of `${mappingTargetType}`. 
The second one will create something like `java.util.List<org.mapstruct.ap.test.callbacks.returning.AttributeDto> target` and the first one creates `List<AttributeDto>`. Using the model one uses the `Type.ftl`, whereas the other one uses the `Type#toString()`

As a side note for us. We can maybe use `<@compress single_line=true>` here to have a more readable template 😄